pkg/middleware/middleware_test.go

```diff
@@ -287,6 +285,21 @@ func TestMiddlewareContext(t *testing.T) {
 					So(sc.resp.Header().Get("Set-Cookie"), ShouldEqual, expectedCookie.String())
 				})
 			}
+
+			Convey("Should not set cookie with setting.SameSite", func() {
+				setting.CookieSameSiteDisabled = true
```

Comment:
In order to test this behaviour you need to set also `setting.CookieSameSiteMode` to some valid `SameSite` (e.g. http.SameSiteDefaultMode) value otherwise the default is 0 and in that case `SameSite` is not written anyway because none of [these](https://github.com/golang/go/blob/go1.13.1/src/net/http/cookie.go#L220) `case` statements are true.

Reply:
Not sure I follow. We check if same site is not disabled [here](https://github.com/grafana/grafana/pull/21472/files/4a52840071461d5e40fcac94278d7d8f41b6ef05#diff-8664c49bf50694ac622b7cb861b8dd1fR41-R42) and only then we set the same site attribute on the cookie. Or are you just referring to setting same site mode for the completeness of the test?

Reply:
The latter